src/widgets/Toast/Toast.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useEffect, useRef } from "react";
+import { CSSTransition } from "react-transition-group";
+import styled from "styled-components";
+import { Alert } from "../../components/Alert";
+import { ToastProps } from "./types";
+
+const StyledToast = styled.div`
+  right: 16px;
+  position: fixed;
+  max-width: calc(100% - 32px);
+  transition: all 250ms ease-in;
+  width: 100%;
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    max-width: 400px;
+  }
+`;
+
+const Toast: React.FC<ToastProps> = ({ alert, onRemove, style, ttl, ...props }) => {
+  const timer = useRef<number>();
+  const ref = useRef(null);
+  const removeHandler = useRef(onRemove);
+  const { id, title, description, type } = alert;
+
+  const handleRemove = useCallback(() => removeHandler.current(id), [id, removeHandler]);
+
+  const handleMouseEnter = () => {
+    clearTimeout(timer.current);
+  };
+
+  const handleMouseLeave = () => {
+    timer.current = setTimeout(() => {
+      handleRemove();
+    }, ttl);
+  };
+
+  useEffect(() => {
+    timer.current = setTimeout(() => {
```

Comment:
here, I would check the timer value before, and call clearTimeout if it's not null (or don't call setTimeout). Just a safeguard to avoid a leak since you have severals deps in the effect